packages-exp/messaging-exp/src/helpers/register.ts

```diff
@@ -64,6 +55,38 @@ const WindowMessagingFactory: InstanceFactory<'messaging-exp'> = (
   return messaging;
 };
 
+const WindowMessagingInternalFactory: InstanceFactory<'messaging-exp-internal'> =
+  (container: ComponentContainer) => {
+    maybeThrowWindowError();
+
+    const messaging = _getProvider(
```

Comment:
I think this can just be
```
container.getProvider('messaging-exp').getImmediate() as MessagingService
```
similar to https://github.com/firebase/firebase-js-sdk/blob/4d747b158021766b21f37879054c45ddd17051ff/packages-exp/app-check-exp/src/index.ts#L74

I'm not sure if installations has a reason for doing it the other way.

Reply:
good call.